src/App.jsx

```diff
@@ -14,45 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const ALPHABET_SORTING = 'alphabetically';
+const LENGTH_SORTING = 'length';
+const REVERSE = 'reverse';
+const UNREVERSE = 'unreverse';
+const RESET = 'reset';
+let currentGoods = [...goodsFromServer];
+
+export const App = () => {
+  const [currentSorting, setCurentSorting] = useState('');
+
+  const [currentDirection, setCurrentDirection] = useState('');
+
+  if (currentSorting === RESET) {
+    currentGoods = [...goodsFromServer];
+    setCurrentDirection('');
+    setCurentSorting('');
+  }
+
+  if (currentDirection === UNREVERSE) {
+    currentGoods = [...goodsFromServer];
+  }
+
+  if (currentSorting === ALPHABET_SORTING
+     || currentSorting === LENGTH_SORTING) {
+    currentGoods = [...goodsFromServer];
+
+    currentGoods.sort((good1, good2) => {
+      switch (currentSorting) {
+        case ALPHABET_SORTING: {
+          return good1.localeCompare(good2);
+        }
+
+        case LENGTH_SORTING: {
+          return good1.length - good2.length;
+        }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        default: return 0;
+      }
+    });
+  }
+
+  if (currentDirection === REVERSE && currentSorting !== RESET) {
+    currentGoods.reverse();
```

Comment:
rewrite this logic based on comments about state values
make for it function outside of component which will be accept goodsFromServer, currentSorting and isReversed states and return reordered goods.
and call it in component.